File: Models/RemoteActionRequest.cs
namespace YFRemote.Server.Models;

public sealed record RemoteActionRequest
{
    public string? Type { get; init; }

    public IReadOnlyList<string>? Keys { get; init; }

    public string? Text { get; init; }

    public int? DeltaX { get; init; }

    public int? DeltaY { get; init; }

    public string? Button { get; init; }

    public int? Delta { get; init; }
}

File: android/app/src/main/java/com/yfremote/android/server/models/RemoteActionRequest.kt
package com.yfremote.android.server.models

import kotlinx.serialization.Serializable

// Spiegelt Models/RemoteActionRequest.cs Feld fuer Feld - camelCase passt ohne @SerialName.
@Serializable
data class RemoteActionRequest(
    val requestId: String? = null,
    val type: String? = null,
    val keys: List<String>? = null,
    val text: String? = null,
    val deltaX: Int? = null,
    val deltaY: Int? = null,
    val button: String? = null,
    val delta: Int? = null,
)
